.github/workflows/linting.yml

```diff
@@ -0,0 +1,39 @@
+name: Frontend linting
+
+# on: [push, pull_request, workflow_dispatch]
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+    linting:
+        runs-on: ubuntu-20.04
+        defaults:
+            run:
+                working-directory: ./webgui-new/
+
+        steps:
+            - name: checkout
+              uses: actions/checkout@v3
+
+            - name: setup Node.JS
+              uses: actions/setup-node@v3
+              with:
+                node-version: 20
+            
+            - name: cache npm
+              uses: actions/cache@v3
+              env:
+                cache-name: npm_cache
+              with:
+                path: ~/.npm
+                key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package.json') }}
```

Comment:
This only caches the global packages data I think (`~/.npm` folder), and could be achieved with the `setup-node` action, as described here:
https://github.com/actions/setup-node#caching-global-packages-data

However, in this case the packages will be installed in the local `node_modules` directory? Shouldn't that be cached as well?